src/main/java/com/criteo/vips/enums/VipsImageType.java

```diff
@@ -0,0 +1,51 @@
+/*
+  Copyright (c) 2020 Criteo
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package com.criteo.vips.enums;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum VipsImageType implements IEnumValue {
+    Error(-1),
+    None(0),
+    Setbuf(1),
+    SetbufForeign(2),
+    Openin(3),
+    Mmapin(4),
+    Mmapinrw(5),
+    Openout(6),
+    Partial(7);
+
+    private int value;
+    private static Map map = new HashMap<>();
```

Comment:
Shouldn't you type the HashMap?

Reply:
My compiler didn't claim about that, but I can type it.